homeassistant/components/scrape/__init__.py

```diff
@@ -70,17 +72,25 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
                 seconds=resource_config.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
             ),
         )
-        await coordinator.async_refresh()
-        if coordinator.data is None:
-            raise PlatformNotReady
 
+        refresh_coroutines.append(coordinator.async_refresh())
+
+        sensors: list[ConfigType] = resource_config.get(SENSOR_DOMAIN, [])
         for sensor_config in sensors:
-            discovery.load_platform(
-                hass,
-                Platform.SENSOR,
-                DOMAIN,
-                {"coordinator": coordinator, "config": sensor_config},
-                config,
+            load_coroutines.append(
+                discovery.async_load_platform(
+                    hass,
+                    Platform.SENSOR,
+                    DOMAIN,
+                    {"coordinator": coordinator, "config": sensor_config},
+                    config,
+                )
             )
 
+    if refresh_coroutines:
```

Comment:
What is the point of this refresh if we do it anyway on platform load?
I think maybe platform load should check if there is a value before calling a refresh.

Reply:
Sorry. I thought about it again overnight.
Maybe we should move more to the platform...
Sending the list of sensors for each coordinator, instead of each sensor individually.
The we can use a single PlatformNotReady as originally intended for the whole list.

Reply:
I think that makes sense and looking at it this morning it's not possible to avoid the 3rd call as in the other comment.
Should we do that here?

Reply:
Yes let's do that here. We're still fixing PlatformNotReady

Reply:
Have a look. I think it looks also cleaner as this.